mono/mini/il2cpp-stubs.cpp

```diff
@@ -873,6 +874,10 @@ Il2CppMonoString* il2cpp_mono_ldstr_checked(Il2CppMonoDomain* domain, Il2CppMono
 Il2CppMonoObject* il2cpp_mono_runtime_try_invoke(Il2CppMonoMethod* method, void* obj, void** params, Il2CppMonoObject** exc, MonoError* error)
 {
 	mono_error_init(error);
+
+	if (((MethodInfo*)method)->declaring_type->valuetype)
+		obj = static_cast<Il2CppObject*>(obj) - 1;
```

Comment:
whats this doing?

Reply:
When il2cpp invokes on a value type we expect the type will be boxed and there will be an object header there that we move the pointer past.  This isn't the case when invoking from the debugger, so we have to make this adjustment.

This code is shamelessly stolen from il2cpp_runtime_invoke(), which does the same thing.